src/main/java/sample/ClientCredentialsAuthenticationFilter.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Base64;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * A filter to perform client authentication for the Token Endpoint, specifically for the Client Credentials Grant.
+ *
+ * See <a href="https://tools.ietf.org/html/rfc6749#section-2.3.1">RFC-6749 2.3.1</a>.
+ */
+public class ClientCredentialsAuthenticationFilter implements Filter {
+
+    private Charset credentialsCharset = UTF_8;
+
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+
+        HttpServletRequest req = (HttpServletRequest) request;
+        if (req.getRequestURI().equals("/oauth/token")) {
+            String grantType = req.getParameter("grant_type");
+            if ("client_credentials".equals(grantType)) {
```

Comment:
Please remove the `grant_type` check. We should authenticate if the target is the token endpoint and there are credentials in the request.